### DISCUSSION TITLE
Is it possible to have the annotation show through the chart

### DISCUSSION THREAD

**Discussion by kenperregaux** [chartjs/chartjs-plugin-annotation#779] [Q&A] [answered]
I have been able to get box annotations working properly. However, I am using the fill of a line chart and would like the annotation to show through the chart like the following (which is written in straight D3).

![image](https://user-images.githubusercontent.com/24940607/182452676-9cd8cc67-96bc-4807-8a17-5dac5688fc8a.png)

**Comment by stockiNail** [comment, marked as answer]
@kenperregaux Foolowing your info, I have created a codepen: https://codepen.io/stockinail/pen/wvmyxBz

Please have a look to:

- `drawTime` in box annotation
- `segment` configuration in order to have a dataset partially filled (as in the picture)

**Reply by kenperregaux** [reply]
Thanks so much this is exactly what I was looking for!

**Reply by kenperregaux** [reply]
Sorry but I tried this and if I use a solid color (e.g. not rgba) it doesn't work. Does it have to be semitransparent?

**Reply by stockiNail** [reply]
Yes, it does otherwise you can not see the box annotation below the dataset. 
Maybe you want to have the box annotation always on top of the dataset. If yes, you can change `drawTime` , using `afterDatasetsDraw`.

**Reply by kenperregaux** [reply]
OK thanks so much!
